### DISCUSSION TITLE
Does it support encryption of networks in existing clusters?

### DISCUSSION THREAD

**Discussion by pluser** [rook/rook#13584] [Q&A] [answered]
Could you tell me whether it is supported to change the fields under `spec.network.connections` after cluster creation?
I'd like to change `connections.encryption.enable` to `true` but found the following caution in the official doc.

https://github.com/rook/rook/blob/bc78b6cad21c62da1501dd19b04d954a51079afe/Documentation/CRDs/Cluster/ceph-cluster-crd.md?plain=1#L223-L225

However, this description has existed before `connections` field has introduced and the meaning of `connections` field is different from other fields in `network` field very much. Changing the fields under `connections` results in the execution of `ceph config`. This is a normal operation for admins. 

FYI, I verified setting `connections.encryption.enable: true` did not cause any problems in my cluster.

**Comment by travisn** [comment]
As I recall, changing the `network.connection.encryption.enabled` setting may cause issues with existing mounted volumes. @Madhu-1 ?

**Reply by pluser** [reply]
I tried it in my local environment and found that existing PVCs were writable even after changing the encryption options. It was also confirmed the communication was encrypted by recreating and remounting the PVCs.
What do you think @Madhu-1 ?

**Reply by Madhu-1** [reply]
Can you please also check that mon is switched to `v2` port after this change and have you also tested cephFS and rbd PVC both?

**Reply by pluser** [reply]
@Madhu-1 
According to the packet capture results, it appears that the v2 messaging protocol is used in the test environment, as port 3300 is used. This was not related to whether or not encryption was enabled.

Also, I did not test for CephFS; I only checked for RBD and RGW.

**Reply by Madhu-1** [reply]
@pluser Thank you, for RBD to work as the Rook set the default for rbd map options for CephFS csi sets the kernel mount options, its good to test cephfs/RBD to ensure the support works as expected.

**Reply by pluser** [reply]
> Thank you, for RBD to work as the Rook set the default for rbd map options for CephFS csi sets the kernel mount options, its good to test cephfs/RBD to ensure the support works as expected.

@Madhu-1 Should I test CephFS case to state the caution that I mentioned doesn't apply to `encryption` field?

**Reply by pluser** [reply]
@Madhu-1 
So, Final conclusion is that encryption is supported for networks in existing clusters, correct?
If you mean that CephFS is exception, we can verify this.

**Reply by Madhu-1** [reply, marked as answer]
AFAIK yes it can be supported.
